modules/server/src/utils/getAllData.js

```diff
@@ -81,10 +81,10 @@ export default async ({
 							...(Object.entries(query).length ? { query } : {}),
 						},
 					})
-					.then(({ body }) => mapHits(body));
+					.then(({ body }) => body.hits.hits);
 				console.timeEnd(timerLabel);
 
-				stream.write({ hits, total });
+				stream.write({ hits: hits.map((hit) => hit?._source), total });
```

Comment:
the changes described here serve the exact same purpose as that `mapHits` function.
what's the trick here? I don't see it.

Reply:
Moving the `mapHits` logic out of the variable `hits` will leave the attributes required for sorting to be used in the next cycle by `previousHits`